### ISSUE TITLE
[BUG] Filename regex in filecontent conditions not always working

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#489]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0

### Priority

Critical

### Current Behavior

Currently, a rule like this:
```yaml
  when:
    builtin.filecontent:
      filePattern: .*\.(xml|java)
      pattern: '$\{VAULT::'
```
won't work because [we're passing the filename pattern directly to `grep`](https://github.com/konveyor/analyzer-lsp/blob/main/provider/internal/builtin/service_client.go#L88).

We might need to expand these "or" conditions into multiple filenames, or create multiple rules. At the moment, the filtering [is done afterwards](https://github.com/konveyor/analyzer-lsp/blob/main/provider/internal/builtin/service_client.go#L113), but it won't get there.

### Expected Behavior

The filecontent condition should be able to handle patterns in the `filePattern` field, and try to match against files with matching names only.

### How Reproducible

Always (Default)

### Steps To Reproduce

See windup-shim test execution, specifically test `legacy-vault-00010-test` should pass.

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by pranavgaikwad** [comment]
@jmle I don't think we are passing the filename pattern to grep, we are passing the search pattern to grep and in the results we get we are filtering-in the results that match the file name pattern, excluding everything else. Am I missing something?

**Comment by jmle** [comment]
@pranavgaikwad you're completely right, I'm closing this for the moment
